_guides/server/echo.md

```diff
@@ -260,7 +264,7 @@ impl Service for Echo {
 #     type Request = Request;
 #     type Error = hyper::Error;
     type Future = Either<
-        FutureResult<Self::Response, Self::Error>,
+        Box<Future<Item=Self::Response, Error=Self::Error>>,
```

Comment:
Yea I agree with you, the `Either` stuff is ugly, especially for a newcomer. I'd make this just a boxed future also.